### PR TITLE
added `refreshAfterWrite` configuration property and use it for caffeine cache impl

### PR DESCRIPTION
**Pull request by snrmwg**

### CODE REVIEW

File: cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java [hunk 1]
```diff
@@ -106,6 +107,13 @@ public Optional<Duration> getExpireAfterWrite() {
         return Optional.ofNullable(expireAfterWrite);
     }
 
+    /**
+     * @return The duration after an entry is eligible for refresh
+     */
+    public Optional<Duration> getRefreshAfterWrite() {
```
Comment (graemerocher) [inline cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java:113]: See caffeine specific, can the new getter/setter go in the caffeine specific subclass?

**Comment by ben-manes** [comment]
This feature requires a CacheLoder (a `LoadingCache`), but this factory does not create that cache type (manually loading). I do not know if that could be wrangled to be made to work, e.g. if the interceptor was resolved early enough to provide the loader rather than late to supply as in `get(key, mappingFunction)`.

**Comment by snrmwg** [comment]
> This feature requires a CacheLoder (a `LoadingCache`), but this factory does not create that cache type (manually loading). I do not know if that could be wrangled to be made to work, e.g. if the interceptor was resolved early enough to provide the loader rather than late to supply as in `get(key, mappingFunction)`.

Somehow missed that requirement, must have been too late last night :)

I was just looking for a solution that would deliver the entry from the cache even if it has expired, but also take care of updating the entry in another thread. That will then probably be the task of the loader. 

I will see to what extent this can be integrated into the Micronaut layer.

**Comment by sdelamo** [comment]
@snrmwg should we close this draft pr?

**Comment by snrmwg** [comment]
yes, please delete it.

Am Do., 12. Okt. 2023 um 10:17 Uhr schrieb Sergio del Amo <
***@***.***>:

**Pull request closed without merging**